app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,78 @@ const validClusterAuth = async (me, queryName, context) => {
   }
 };
 
-var getAllowedChannels = async(me, org_id, action, field, context)=>{
+// get allowed resources
+const getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
   const { models } = context;
-  var channels = await models.Channel.find({ org_id });
-  return await filterChannelsToAllowed(me, org_id, action, field, channels, context);
-};
+  let allAllowed = false;
+  let resources;
 
-var filterChannelsToAllowed = async(me, org_id, action, field, channels, context)=>{
-  const { models } = context;
-  var decisionInputs = _.map(channels, (channel)=>{
-    return {
-      type: field,
-      action,
-      uuid: channel.uuid,
-      name: channel.name,
-    };
-  });
-  var decisions = await models.User.isAuthorizedBatch(me, org_id, decisionInputs, context);
-  channels = _.filter(channels, (val, idx)=>{
-    return decisions[idx];
-  });
-  return channels;
-};
+  try {
+    await validAuth(me, org_id, action, type, queryName, context);
+    allAllowed = true;
+  }
+  catch(e){ // if exception thrown, user does NOT have auth to all resources of this type, and code must later filter based on fine grained auth
+  }
 
-var getAllowedSubscriptions = async(me, org_id, action, field, context)=>{
-  const { models } = context;
-  var subscriptions = await models.Subscription.find({ org_id });
-  return await filterSubscriptionsToAllowed(me, org_id, action, field, subscriptions, context);
+  const modelType = type.charAt(0).toUpperCase() + type.slice(1);
+
+  // find by resource type
+  if (searchByTags) {
+    resources = await models[modelType].find({org_id, tags: {$all: searchByTags}});
+  }
+  else if (type === 'group') {
+    if (searchByUuid) {
+      resources = await models[modelType].find({org_id, uuid: {$in: searchByUuid}});
+    }
+    else {
+      resources = await models[modelType].find({org_id}).lean({virtuals: true});
+    }
+  }
+  else if (type === 'subscription') {
+    if (searchQuery) {
+      resources = await models[modelType].find(searchQuery).lean({virtuals: true});
+    }
+  }
+  else {
+    resources = await models[modelType].find({org_id});
+  }
```

Comment:
Why do some queries use `.lean({virtuals: true})` but others do not?  Is this intentional?

Reply:
I understand that .lean({virtuals: true}) can retrieve a javascript object over a mongoose document with any included virtuals. I don't think we actually use those virtuals, so I'm unsure why they were written there. Any thoughts?

Reply:
I think you're right, and we never / almost never use the 'mongoose virtuals'.  Sorting out that entire mess is beyond the scope of this PR, so "taken from how they were written previously" is an excellent answer.

Resolved.